File: #1475.final-prices-with-a-special-discount-in-a-shop.cpp
class Solution {
public:
    vector<int> finalPrices(vector<int>& prices) {

        int size = prices.size();
        for (int i = 0; i < size; i++) {
            int j;
            for (j = i+1; j < size; j++) {
                if (prices[j] <= prices[i]) {
                    break;
                }
            }

            if (j < size && prices[j] <= prices[i]) {
                prices[i] -= prices[j];
            }
        }
        return prices;
    }
};

File: #1475.final-prices-with-a-special-discount-in-a-shop.js
/**
 * @param {number[]} prices
 * @return {number[]}
 */
var finalPrices = function(prices) {
	return prices.map((price, index) => {
		return price - (prices.slice(index + 1).find(dis => dis <= price) || 0);
	})
};
